### PR TITLE
Custom recognize service domain 

### PR DESCRIPTION
**Pull request by wangrui-msft**
# Add new recognize service updates
1. It adds new reason code for recognize action
2. It adds new custom recognize service domain
3. It updates the models of play and recognize

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationClient.cs [hunk 1]
```diff
@@ -215,6 +215,11 @@ private static AnswerCallRequestInternal CreateAnswerCallRequest(AnswerCallOptio
             }
 
             AnswerCallRequestInternal request = new AnswerCallRequestInternal(options.IncomingCallContext, options.CallbackUri.AbsoluteUri);
+            // Add custom cognitive service domain name
+            if (string.IsNullOrEmpty(options.AzureCognitiveServiceDomainName))
```
Comment (williamzhao87) [inline sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationClient.cs:219]:
```suggestion
            if (!string.IsNullOrEmpty(options.AzureCognitiveServiceDomainName))
```

File: sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationClient.cs [hunk 2]
```diff
@@ -504,6 +509,11 @@ private static CreateCallRequestInternal CreateCallRequest(CreateCallOptions opt
                 options.Targets.Select(t => CommunicationIdentifierSerializer.Serialize(t)),
                 sourceDto,
                 options.CallbackUri.AbsoluteUri);
+            // Add custom cognitive service domain name
+            if (string.IsNullOrEmpty(options.AzureCognitiveServiceDomainName))
```
Comment (williamzhao87) [inline sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationClient.cs:513]:
```suggestion
            if (!string.IsNullOrEmpty(options.AzureCognitiveServiceDomainName))
```

Comment (wangrui-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationClient.cs:513]: ah, good catch, will correct it

File: sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationClient.cs (hunk 1 above)
Comment (wangrui-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationClient.cs:219]: good catch, will correct it.

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/CreateCallOptions.cs [hunk 3]
```diff
@@ -55,5 +55,10 @@ public CreateCallOptions(CallSource callSource, IEnumerable<CommunicationIdentif
         /// Repeatability Headers.
         /// </summary>
         public RepeatabilityHeaders RepeatabilityHeaders { get; set; }
+
+        /// <summary>
+        /// The identifier of the Cognitive Service resource assigned to this call.
```
Comment (williamzhao87) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/CreateCallOptions.cs:60]: On service side, I think it's been updated:
/// The endpoint URL of the Azure Cognitive Services resource attached

https://skype.visualstudio.com/SCC/_git/sync_calling_concore-platform/pullrequest/720937?_a=files&path=/Microsoft.Skype.Platform/Microsoft.Skype.Platform.ExecutionAgent/Azure.Communication.Service.ServerCalling/Contracts/V4-2023-01-15-preview/Requests/AnswerCallRequest.cs

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/AnswerCallOptions.cs [hunk 4]
```diff
@@ -41,5 +41,10 @@ public AnswerCallOptions(string incomingCallContext, Uri callbackUri)
         /// Repeatability Headers.
         /// </summary>
         public RepeatabilityHeaders RepeatabilityHeaders { get; set; }
+
+        /// <summary>
+        /// The identifier of the Cognitive Service resource assigned to this call.
```
Comment (williamzhao87) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/AnswerCallOptions.cs:46]: On service side, I think it's been updated:
/// The endpoint URL of the Azure Cognitive Services resource attached

https://skype.visualstudio.com/SCC/_git/sync_calling_concore-platform/pullrequest/720937?_a=files&path=/Microsoft.Skype.Platform/Microsoft.Skype.Platform.ExecutionAgent/Azure.Communication.Service.ServerCalling/Contracts/V4-2023-01-15-preview/Requests/AnswerCallRequest.cs

Comment (wangrui-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/AnswerCallOptions.cs:46]: will correct the comments

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/CreateCallOptions.cs (hunk 3 above)
Comment (wangrui-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/CreateCallOptions.cs]: will correct the comment

Comment (wangrui-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/CreateCallOptions.cs]: will correct the comment

**Review by williamzhao87 — APPROVED** [review]

**Pull request merged**